utils/buffer/src/slice.rs

```diff
@@ -182,6 +357,18 @@ impl SharedState {
             .unwrap();
     }
 
+    // Toggles the bit at the specified `position` in the `SharedState`.
+    //
+    // After a message held by a buffer slice has been processed, the corresponding bit in the
+    // shared state is toggled (flipped). When the shared state for a given region reaches zero
+    // (i.e., all bits are cleared), the buffer pool knows it can safely reclaim or reuse that
+    // memory slice.
+    //
+    // Uses atomic bitwise operations to ensure thread-safe toggling without locks. It manipulates
+    // the shared state in-place using the `AtomicU8::fetch_update` method, which atomically
+    // applies a bitwise XOR (`^`) to toggle the bit at the specified `position`.
+    //
+    // Panics if the `position` is outside the range of 1-8, as this refers to an invalid bit.
```

Comment:
Why did you use  normal comments and not public docs ones here?

Reply:
It's private.